service/external_paysession.go

```diff
@@ -1,78 +1,46 @@
 package service
 
 import (
-	"encoding/json"
 	"fmt"
 	"net/http"
-	"strconv"
-	"strings"
 
 	"github.com/companieshouse/chs.go/log"
 	"github.com/companieshouse/payments.api.ch.gov.uk/models"
-	"github.com/gorilla/mux"
 )
 
 // CreateExternalPaymentJourney creates an external payment session with a Payment Provider that is given, e.g: GovPay
-func (service *PaymentService) CreateExternalPaymentJourney(w http.ResponseWriter, req *http.Request) {
-	vars := mux.Vars(req)
-	id := vars["payment_id"]
-	if id == "" {
-		log.ErrorR(req, fmt.Errorf("payment id not supplied"))
-		w.WriteHeader(http.StatusBadRequest)
-		return
-	}
-
-	paymentSession, httpStatus, err := service.GetPaymentSession(id)
-	if err != nil {
-		w.WriteHeader(httpStatus)
-		log.ErrorR(req, err)
-		return
-	}
+func (service *PaymentService) CreateExternalPaymentJourney(req *http.Request, paymentSession *models.PaymentResourceRest) (*models.ExternalPaymentJourney, int, error) {
```

Comment:
Service layer shouldn't decide the status code to return so shouldn't require the `int` return argument directly here.

 Rather than returning a status code we should explore the same kind of pattern used in java land for company accounts that gives a service level response status ( https://github.com/companieshouse/company-accounts.api.ch.gov.uk/blob/develop/src/main/java/uk/gov/companieshouse/api/accounts/service/response/ResponseStatus.java ) alongside the return object/struct that we are dealing with